frontend/packages/ux-editor/src/components/rightMenu/DynamicsTab.tsx

```diff
@@ -0,0 +1,112 @@
+import { ExpressionElement } from "../config/expressions/ExpressionContent";
+import { ExpressionPropertyBase, ExpressionPropertyForGroup } from "../../types/Expressions";
+import { LayoutItemType } from "../../types/global";
+
+export interface Dynamic {
+  id?: string;
+  editMode: boolean;
+  property?: ExpressionPropertyBase | ExpressionPropertyForGroup; // action?
+  expressionElements?: ExpressionElement[];
+}
+
+export const DynamicsTab = () => {
+  const { form, formId } = useContext(FormContext);
+
+  if (!formId || !form) return null;
+
+  const defaultDynamic: Dynamic = { editMode: true, expressionElements: [] };
+  const [dynamics, setDynamics] = React.useState<Dynamic[]>([defaultDynamic]); // default state should be already existing dynamics
+  const [showAddDynamicButton, setShowAddDynamicButton] = React.useState<boolean>(false);
+  const [showRemoveDynamicButton, setShowRemoveDynamicButton] = React.useState<boolean>(false);
+  const t = useText();
+
+  // adapt list of actions if component is group
+  const expressionProperties = form.itemType === LayoutItemType.Container ?
+    (Object.values(ExpressionPropertyBase) as string[])
+      .concat(Object.values(ExpressionPropertyForGroup) as string[]) : Object.values(ExpressionPropertyBase);
+
+  const addDynamic = () => {
+    // Convert dynamic object to correct format and save dynamic to layout with api call
+    // Set editMode fields for all prev dynamics to false
+    const dynamic: Dynamic = { editMode: true, expressionElements: [] };
+    setDynamics(prevDynamics => prevDynamics.map(prevDynamic => ({ ...prevDynamic, editMode: false })).concat(dynamic));
+  };
+
+  const editDynamic = (dynamic: Dynamic) => {
+    // Convert dynamic object to correct format and save dynamic to layout with api call
+    // Set editMode fields for all prev dynamics to false
+    const updatedDynamics = dynamics.map(prevDynamic => {
+      if (prevDynamic === dynamic) return { ...prevDynamic, editMode: true }
+    else return { ...prevDynamic, editMode: false } });
+
+    setDynamics([...updatedDynamics]);
+  };
+
+  const removeDynamic = (dynamic: Dynamic) => {
+    // Convert dynamic object to correct format and save dynamic to layout with api call
+    // Set editMode fields for all prev dynamics to false
+    if (dynamics.length === 1) {
+      setDynamics(prevDynamics => prevDynamics.filter(prevDynamic => prevDynamic !== dynamic).concat(defaultDynamic));
+    }
+    else {
+      setDynamics(prevDynamics => prevDynamics.filter(prevDynamic => prevDynamic !== dynamic));
+    }
+  };
+
+  const getProperties = () => {
+    const alreadyUsedProperties = dynamics.map(dynamic => dynamic.property) as string[];
+    const availableProperties = expressionProperties.filter(expressionProperty => !Object.values(alreadyUsedProperties).includes(expressionProperty));
+    return { availableProperties, expressionProperties }
+  };
+
+  useEffect(() => {
```

Comment:
This hook call should be moved up to the others, before the `if (!formId || !form) return null;` line.